using-timescaledb/limitations.md

```diff
@@ -0,0 +1,42 @@
+# Limitations [](limitations)
+
+While TimescaleDB generally offers capabilities that go beyond what
+PostgreSQL offers, there are some limitations to using hypertables,
+and, in particular, distributed hypertables. This section documents
+the common limitations when using both regular and distributed
+hypertables.
+
+## Hypertable Limitations [](hypertable-limitations)
+
+- Foreign key constraints referencing a hypertable are not supported.
+- Time dimensions (columns) used for partitioning cannot have NULL
+  values.
+- Unique indexes must include all columns that are partitioning
+  dimensions.
+
+## Distributed Hypertable Limitations [](distributed-hypertable-limitations)
+
+All the limitations of regular hypertables also apply to distributed
+hypertables. In addition, the following limitations apply specifically
+to distributed hypertables:
+
+- Background job scheduling is not supported.
+- Continuous aggregates are not supported.
+- Compressed hypertables are not supported.
+- Tablespaces cannot be attached to a distributed hypertable on the
+  access node. It is still possible attach tablespaces on data nodes.
+- Roles and permissions are assumed to be consistent across the nodes
+  of a distributed database, but consistency is not enforced.
+- Tables referenced by foreign key constrains in a distributed
+  hypertable must be present on all data nodes.
```

Comment:
To be clear, we don't support foreign keys at all if the tables don't exist on data nodes, so I don't think changing the order will improve.

The issue is that the CREATE TABLE on the data nodes will fail if the referenced table is not present.

Reply:
The current formulation requires to have it on all data nodes, while I guess it needs to be on access node too, otherwise create table will fail, isn't it?
IMHO moving this limitation after the join limitation will help users a lot.

Reply:
Yes, it needs to be on the access node, which was assumed. But I will clarify. I can change the order of limtitations too if you feel strongly but since foreign keys are created before they can be used in joins I don't think the proposed order is an improvement. Further, foreign keys are not a prerequisite for joins although they can be used.